core/src/main/java/org/jruby/ext/ffi/jffi/DynamicLibrary.java

```diff
@@ -60,8 +72,25 @@ final Library getNativeLibrary() {
     @JRubyMethod(name = {  "open" }, meta = true)
     public static final  IRubyObject open(ThreadContext context, IRubyObject recv, IRubyObject libraryName, IRubyObject libraryFlags) {
         final String libName = libraryName.isNil() ? null : libraryName.toString();
+        String loadName;
+        if (libName.contains(":") && !new File(libName).isAbsolute()) { // windows is C:\\ ..., but otherwise we may have an URI
+            // Use internal logic to parse the URI
+            FileResource resource = JRubyFile.createResource(context, libName);
+            if (JRubyFile.isResourceRegularFile(resource))
+            {
+                loadName = resource.absolutePath();
+            }
+            else try (InputStream internalStream = resource.openInputStream()){
+                loadName = extractLibrary(resource.path(), internalStream);
+            } catch (IOException e) {
+                // let it fail normally, file not found
+                loadName = resource.absolutePath();
+            }
+        }
+        else
```

Comment:
no single line if or else without curly brackets unless they are on same line (e.g. if (foo) something;).  This will never apply to if/else cases.  If both then they need curly brackets.